cv::cuda::resize failed with Continuous GpuMat

### System Information

OpenCV version: 4.11.0 
Operating System / Platform: Ubuntu 22.04 
Compiler & compiler version: GCC 11.4.0

Docker Image: nvidia/cuda:11.8.0-cudnn8-devel-ubuntu22.04

```
root@031cee7bf3ff:~# nvidia-smi
Wed Jan 14 11:15:47 2026
+-----------------------------------------------------------------------------------------+
| NVIDIA-SMI 570.195.03             Driver Version: 570.195.03     CUDA Version: 12.8     |
|-----------------------------------------+------------------------+----------------------+
| GPU  Name                 Persistence-M | Bus-Id          Disp.A | Volatile Uncorr. ECC |
| Fan  Temp   Perf          Pwr:Usage/Cap |           Memory-Usage | GPU-Util  Compute M. |
|                                         |                        |               MIG M. |
|=========================================+========================+======================|
|   0  NVIDIA GeForce RTX 3060        Off |   00000000:00:08.0 Off |                  N/A |
|  0%   41C    P0             29W /  170W |       0MiB /  12288MiB |      3%      Default |
|                                         |                        |                  N/A |
+-----------------------------------------+------------------------+----------------------+

+-----------------------------------------------------------------------------------------+
| Processes:                                                                              |
|  GPU   GI   CI              PID   Type   Process name                        GPU Memory |
|        ID   ID                                                               Usage      |
|=========================================================================================|
|  No running processes found                                                             |
+-----------------------------------------------------------------------------------------+
```


### Detailed description

When `cv::cuda::resize`  with a continuous GpuMat, get errors:

```
terminate called after throwing an instance of 'cv::Exception'
  what():  OpenCV(4.11.0) /home/opencv/opencv_contrib/modules/cudev/include/opencv2/cudev/ptr2d/texture.hpp:147: error: (-217:Gpu API call) invalid argument in function 'createTextureObject'
```


### Steps to reproduce

```cpp
#include <opencv2/opencv.hpp>
#include <opencv2/cudawarping.hpp>

int main() {
    cv::cuda::GpuMat gpu_mat(10, 10, CV_8U);
    cv::cuda::GpuMat gpu_mat_out;
    cv::cuda::resize(gpu_mat, gpu_mat_out, cv::Size(100, 100), 0.0, 0.0, cv::INTER_NEAREST);
    std::cout << "OK Here\n";

    gpu_mat = cv::cuda::createContinuous(10, 10, CV_8U);
    cv::cuda::resize(gpu_mat, gpu_mat_out, cv::Size(100, 100), 0.0, 0.0, cv::INTER_NEAREST);
    std::cout << "Can't get here\n";

    return 0;
}
```

### Issue submission checklist

- [x] I report the issue, it's not a question
- [x] I checked the problem with documentation, FAQ, open issues, forum.opencv.org, Stack Overflow, etc and have not found any solution
- [x] I updated to the latest OpenCV version and the issue is still there
- [x] There is reproducer code and related data files (videos, images, onnx, etc)

--- a/modules/cudawarping/test/test_resize.cpp
+++ b/modules/cudawarping/test/test_resize.cpp
@@ -261,5 +261,59 @@ INSTANTIATE_TEST_CASE_P(CUDA_Warping, ResizeTextures, testing::Combine(
     testing::Values(Interpolation(cv::INTER_NEAREST), Interpolation(cv::INTER_LINEAR), Interpolation(cv::INTER_CUBIC))));
 
 
+
+// Test for issue #28407: cv::cuda::resize fails with continuous GpuMat
+PARAM_TEST_CASE(ResizeContinuous, cv::cuda::DeviceInfo, cv::Size, MatType, Interpolation)
+{
+    cv::cuda::DeviceInfo devInfo;
+    cv::Size size;
+    int type;
+    int interpolation;
+
+    virtual void SetUp()
+    {
+        devInfo = GET_PARAM(0);
+        size = GET_PARAM(1);
+        type = GET_PARAM(2);
+        interpolation = GET_PARAM(3);
+
+        cv::cuda::setDevice(devInfo.deviceID());
+    }
+};
+
+CUDA_TEST_P(ResizeContinuous, Regression_Issue28407)
+{
+    // Test both regular and continuous GpuMat
+    cv::Mat src = randomMat(size, type);
+
+    // Test 1: Regular GpuMat (should work)
+    cv::cuda::GpuMat d_src_regular(src);
+    cv::cuda::GpuMat d_dst_regular;
+    EXPECT_NO_THROW(
+        cv::cuda::resize(d_src_regular, d_dst_regular, cv::Size(size.width * 2, size.height * 2), 0, 0, interpolation)
+    );
+
+    // Test 2: Continuous GpuMat (this is the bug - should work but currently fails)
+    cv::cuda::GpuMat d_src_continuous = cv::cuda::createContinuous(size.height, size.width, type);
+    d_src_continuous.upload(src);
+    cv::cuda::GpuMat d_dst_continuous;
+    EXPECT_NO_THROW(
+        cv::cuda::resize(d_src_continuous, d_dst_continuous, cv::Size(size.width * 2, size.height * 2), 0, 0, interpolation)
+    );
+
+    // Verify both produce the same result
+    cv::Mat h_dst_regular, h_dst_continuous;
+    d_dst_regular.download(h_dst_regular);
+    d_dst_continuous.download(h_dst_continuous);
+
+    EXPECT_MAT_NEAR(h_dst_regular, h_dst_continuous, 0.0);
+}
+
+INSTANTIATE_TEST_CASE_P(CUDA_Warping, ResizeContinuous, testing::Combine(
+    ALL_DEVICES,
+    testing::Values(cv::Size(10, 10), cv::Size(64, 64), cv::Size(128, 128)),
+    testing::Values(MatType(CV_8UC1), MatType(CV_8UC3), MatType(CV_32FC1)),
+    testing::Values(Interpolation(cv::INTER_NEAREST), Interpolation(cv::INTER_LINEAR), Interpolation(cv::INTER_CUBIC))));
+
 }} // namespace
 #endif // HAVE_CUDA

--- a/modules/cudev/include/opencv2/cudev/ptr2d/texture.hpp
+++ b/modules/cudev/include/opencv2/cudev/ptr2d/texture.hpp
@@ -175,8 +175,11 @@ namespace cv {  namespace cudev {
             texRes.resType = cudaResourceTypePitch2D;
             texRes.res.pitch2D.height = rows;
             texRes.res.pitch2D.width = cols;
-            // temporary fix for single row/columns until TexturePtr is reworked
-            if (rows == 1 || cols == 1) {
+            const size_t minPitch = cols * sizeof(T1);
+            const size_t alignment = 32;
+            const bool isPitchAligned = (step % alignment == 0);
+            const bool needsReallocation = (rows == 1) || (cols == 1) || (step < minPitch) || (!isPitchAligned);
+            if (needsReallocation) {
                 size_t dStep = 0;
                 CV_CUDEV_SAFE_CALL(cudaMallocPitch(&internalSrc, &dStep, cols * sizeof(T1), rows));
                 CV_CUDEV_SAFE_CALL(cudaMemcpy2D(internalSrc, dStep, data, step, cols * sizeof(T1), rows, cudaMemcpyDeviceToDevice));

